examples/cpp/HelloWorldExample/Monitor.cpp

```diff
@@ -0,0 +1,334 @@
+        topic_datawriters,                                           // Source entities
+        topic_datareaders,                                           // Target entities
+        n_bins_,                                                     // Number of bins
+        now - std::chrono::seconds(t_interval_),                     // t_from
+        now,                                                         // t_to
+        StatisticKind::MEAN);                                        // Statistic
+
+    for (auto latency : latency_data)
+    {
+
+        std::cout << "Fast DDS Latency of HelloWorld topic: ["
+                  << timestamp_to_string(latency.first) << ", " << latency.second / 1000 << " μs]" << std::endl;
+    }
+
+    return latency_data;
+}
+
+std::vector<StatisticsData> Monitor::get_publication_throughput_mean()
+{
+    std::vector<StatisticsData> publication_throughput_data{};
+
+    std::vector<EntityId> participants = StatisticsBackend::get_entities(EntityKind::PARTICIPANT);
+    EntityId participant_id = -1;
+    Info participant_info;
+    for (auto participant : participants)
+    {
+        participant_info = StatisticsBackend::get_info(participant);
+        if (participant_info["name"] == "Participant_pub")
+        {
+            participant_id = participant;
+        }
+    }
+
+    if (participant_id < 0)
+    {
+        return publication_throughput_data;
+    }
+
+    /* Get the DataWriters and DataReaders in a Topic */
+    std::vector<EntityId> topic_datawriters = StatisticsBackend::get_entities(
+        EntityKind::DATAWRITER,
+        participant_id);
+
+    /* Get the current time */
+    std::chrono::system_clock::time_point now = std::chrono::system_clock::now();
+
+    /*
+     *
+     */
+    publication_throughput_data = StatisticsBackend::get_data(
+        DataKind::PUBLICATION_THROUGHPUT,                            // DataKind
+        topic_datawriters,                                           // Source entities
+        n_bins_,                                                     // Number of bins
+        now - std::chrono::seconds(t_interval_),                     // t_from
+        now,                                                         // t_to
+        StatisticKind::MEAN);                                        // Statistic
+
+    for (auto publication_throughput : publication_throughput_data)
+    {
+
+        std::cout << "Publication throughput of Participant " << participant_info["name"] << ": ["
+                  << timestamp_to_string(publication_throughput.first) << ", "
+                  << publication_throughput.second << " B/s]" << std::endl;
+    }
+
+    return publication_throughput_data;
+}
+
+/***************************************************************
+* Monitor Listener callbacks implementation
+***************************************************************/
+void Monitor::Listener::on_participant_discovery(
```

Comment:
**In general** 
I would do specific discovery messages for each entity kind
The `id` here is not relevant to the user, but more like a mess. Add GUID in Participant, Writer and Reader, comment that a new topic has been discovered, mention the physical data when it arrives. In summary, use `get_info` with the new entities and give some information